tests/api2/ups.py

```diff
@@ -53,22 +69,62 @@ def test_02_Set_UPS_options():
 
 
 @pytest.mark.parametrize('data', first_ups_list)
-def test_03_look_at_UPS_options_output_of_(data):
+def test_05_look_at_UPS_options_output_of_(data):
     assert payload[data] == results.json()[data], results.text
 
 
-def test_04_get_API_reports_UPS_configuration_as_saved():
+def test_06_starting_ups_service():
+    payload = {
+        "service": "ups",
+        "service-control": {
+            "onetime": True,
+        }
+    }
+    results = POST('/service/start/', payload)
+    assert results.status_code == 200, results.text
+    sleep(1)
+
+
+def test_07_look_UPS_service_status_is_running():
+    results = GET(f'/service/id/{ups_id}/')
+    assert results.status_code == 200, results.text
+    assert results.json()['state'] == 'RUNNING', results.text
+
+
+def test_08_get_API_reports_UPS_configuration_as_saved():
     global results
     results = GET('/ups/')
     assert results.status_code == 200, results.text
 
 
 @pytest.mark.parametrize('data', first_ups_list)
-def test_05_look_API_reports_UPS_configuration_of_(data):
+def test_09_look_API_reports_UPS_configuration_of_(data):
     assert payload[data] == results.json()[data], results.text
 
 
-def test_06_Change_UPS_options():
+def test_10_stop_ups_service():
+    payload = {
+        "service": "ups",
+        "service-control": {
+            "onetime": True,
+        }
+    }
+    results = POST('/service/stop/', payload)
+    assert results.status_code == 200, results.text
+    sleep(1)
```

Comment:
Let's increase this to 30 please. upsmon daemon can take some time to fully exit even if the call returns back. Also we could skip the `sleep(30)` below then

Reply:
test_12 has a sleep 30 second.

Reply:
Yes but we check if the service has stopped before that sleep. So we can sleep after stopping and then check if the service has stopped and then start it / make changes etc. Consolidating the sleep ; )

Reply:
Also you have 2 `test_12` this way : D

Reply:
Yeah I forgot thanks